Core/Graphics/Source/Graphics.cpp

```diff
@@ -276,7 +276,7 @@ namespace Babylon
         }
         {
             std::scoped_lock lock{m_state.Mutex};
-            m_state.Resolution.HardwareScalingLevel = level;
+            m_state.Resolution.HardwareScalingLevel = std::max(level, 1.0f); // Disallow upscaling.
```

Comment:
It seems weird to check that `level` is greater than epsilon above, but then if it is between epsilon and 1 to clamp it to 1 here. To me it would make more sense to just change the check on line 275 to throw if the value is less than 1 with an error message that says upscaling is not supported. I think this will make it more immediately obvious why things are not behaving as expected if you set a level less than 1.

Reply:
Good idea, made the change.